### PR TITLE
add analytics with ApplicationInsights

### PR DESCRIPTION
**Pull request by kkuepper**

### CODE REVIEW

File: plugins/4.applicationInsights.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    router.afterEach((route: { name: any }) => {
+      appInsights.stopTrackPage(route.name);
+    });
+  };
+
+  const appInsights = new ApplicationInsights({
+    config: {
+      connectionString:
+        "InstrumentationKey=45d71410-6840-472a-bf95-45f77749c0a2;IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/;LiveEndpoint=https://westeurope.livediagnostics.monitor.azure.com/",
```
Comment (sifferhans) [inline plugins/4.applicationInsights.ts:72]: This connection-string should perhaps be an environment variable?

File: plugins/4.applicationInsights.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      /* ...Other Configuration Options... */
+    },
+  });
+  appInsights.loadAppInsights();
+  appInsights.trackPageView();
+
+  inject("appInsights", appInsights);
```
Comment (sifferhans) [inline plugins/4.applicationInsights.ts:79]: What is this `inject('appInsights', appInsights)` ?

**Review by sifferhans — APPROVED** [review]

File: plugins/4.applicationInsights.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+// This plugin was inspired by https://github.com/nuxt-community/applicationinsights-module
```
Comment (SimonSimCity) [inline plugins/4.applicationInsights.ts:1]: What's actually the reason for not using the original package ..?

Comment (kkuepper) [reply plugins/4.applicationInsights.ts:1]: It doesn't work with Nuxt 3

File: plugins/4.applicationInsights.ts (hunk 2 above)
Comment (kkuepper) [reply plugins/4.applicationInsights.ts:79]: I assumed that it makes it usable in other components. But I haven't gotten that far yet.
There's a reason it's only a draft PR. I still need to figure out how to provide additional info and how to create custom events.

Comment (sifferhans) [reply plugins/4.applicationInsights.ts:79]: I see! You can return it like this from the plugin:

```ts
return {
	provide: {
		appInsights
	}
}
```

You will then have access to it through the `useNuxtApp()` composable:

```ts
const { $appInsights } = useNuxtApp()
```

Comment (kkuepper) [reply plugins/4.applicationInsights.ts:79]: What's the best way to use appInsights in the 3.mediaPlayer.ts plugin?

**Comment by kkuepper** [comment]
@sifferhans How can I get strong typing for $appInsights? Or should I use a different mechanism?
Screenshot 2023-06-28 at 3 33 32 PM

**Comment by SimonSimCity** [comment]
@kkuepper updating this branch with the latest TypeScript changes (#159) should do half the job. Another part is the file `.nuxt/components.plugin.mjs`, which I'm on... After adding `.nuxt/components.plugin.ts` with an empty object as export the completion worked for me, excepted for in vue-files 😅🙃 trying to fix that as well...

File: plugins/4.applicationInsights.ts (hunk 2 above)
Comment (SimonSimCity) [reply plugins/4.applicationInsights.ts]: @kkuepper https://nuxt.com/docs/guide/directory-structure/plugins#typing-plugins - combined with the statement above the anchor, which is what Sigve was pointing out. Auto completion for those helpers is still tricky 🙈

Injecting it the way you did here (https://vuejs.org/guide/components/provide-inject.html) is the Vue-way, but Nuxt has different way, which is the one Sigve mentioned 🤗

**Comment by SimonSimCity** [comment]
I've squashed the changes, moved the connection string to the environment variable `NUXT_PUBLIC_APPLICATION_INSIGHTS`, moved the extra dependency `nuxt` back as dev-dependency and fixed some additional imports.

**Comment by SimonSimCity** [comment]
It just seems that despite of me adding it as variable in the settings of this project, it doesn't seem to be available in the process, because otherwise the tests of [5817d90](https://github.com/bcc-code/bmm-web/pull/157/commits/5817d90b0e4c83f5a874f8fde469ff3f72252d6d) should not have failed ...

**Review by SimonSimCity — APPROVED** [review]

**Comment by SimonSimCity** [comment]
Fixes #135

**Pull request merged**